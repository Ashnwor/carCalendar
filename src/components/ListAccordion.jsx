import moment from 'moment';
import React from 'react';
import { List, Button, Colors } from 'react-native-paper';
import call from 'react-native-phone-call';
import 'moment/locale/tr';

export default function ListAccordion(props) {
	console.log(props.details);
	const { navigation, day } = props;
	const { setContentToDelete, setDialogVisible } = props.functions;
	const licensePlate = props.licensePlateName;
	const {
		brand,
		model,
		clientNameSurname,
		clientPhone,
		referance,
		givenDate,
		notificationToken,
	} = props.details;
	return (
		<List.Accordion
			title={licensePlate}
			key={licensePlate}
			left={(props) => <List.Icon {...props} icon="car" />}
		>
			<List.Item
				title={`Marka: ${brand}`}
				key={`${licensePlate}-${brand}`}
				left={(props) => <List.Icon {...props} icon="tag-multiple" />}
			/>
			<List.Item
				title={`Model: ${model}`}
				key={`${licensePlate}-${model}`}
				left={(props) => <List.Icon {...props} icon="tag" />}
			/>
			<List.Item
				title={`Müşteri: ${clientNameSurname}`}
				key={`${licensePlate}-${clientNameSurname}`}
				left={(props) => <List.Icon {...props} icon="account" />}
			/>
			<List.Item
				title={`Telefon: ${clientPhone}`}
				key={`${licensePlate}-${clientPhone}`}
				left={(props) => <List.Icon {...props} icon="phone" />}
				right={(props) => <List.Icon {...props} icon="call-made" />}
				onPress={() =>
					call({
						number: clientPhone.replace(' ', ''),
						prompt: false,
					}).catch(console.error)
				}
			/>
			{referance ? (
				<List.Item
					title={`Referans: ${referance}`}
					key={`${licensePlate}-${referance}`}
					left={(props) => <List.Icon {...props} icon="account-multiple" />}
				/>
			) : null}
			<List.Item
				title={`Verilen tarih: ${`${moment(givenDate).format('D MMMM YYYY')}`}`}
				key={`${licensePlate}-${givenDate}`}
				left={(props) => <List.Icon {...props} icon="calendar" />}
			/>
			<List.Item
				key={`${licensePlate}-controls`}
				right={() => (
					<>
						<Button
							uppercase={false}
							mode="outlined"
							icon="pencil"
							style={{ margin: 4 }}
							onPress={() =>
								navigation.navigate('Details', {
									day,
									edit: true,
									editThis: {
										licensePlate,
										brand,
										model,
										clientNameSurname,
										clientPhone,
										referance,
										givenDate,
										notificationToken,
									},
								})
							}
						>
							Düzenle
						</Button>
						<Button
							uppercase={false}
							style={{ margin: 4, width: 90 }}
							color={Colors.red500}
							mode="contained"
							icon="delete"
							onPress={() => {
								setContentToDelete(licensePlate);
								setDialogVisible(true);
							}}
						>
							Sil
						</Button>
					</>
				)}
			/>
		</List.Accordion>
	);
}
