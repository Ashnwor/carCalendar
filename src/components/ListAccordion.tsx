import { StackNavigationProp } from '@react-navigation/stack';
import moment from 'moment';
import React from 'react';
import { List, Button, Colors } from 'react-native-paper';
// @ts-ignore
import call from 'react-native-phone-call';

import { RootStackParamList } from '../CalendarApp';
import 'moment/locale/tr';

type ProfileScreenNavigationProp = StackNavigationProp<RootStackParamList, 'List'>;

type Props = {
	navigation: ProfileScreenNavigationProp;
	day: Record<string, any>;
	licensePlateName: string;
	details: Record<string, any>;
	// eslint-disable-next-line @typescript-eslint/ban-types
	functions: {
		setContentToDelete: (licensePlate: string) => void;
		setDialogVisible: (bool: boolean) => void;
	};
};

const ListAccordion: React.FC<Props> = ({
	navigation,
	day,
	licensePlateName,
	details,
	functions,
}) => {
	const { setContentToDelete, setDialogVisible } = functions;
	const licensePlate = licensePlateName;
	const {
		brand,
		model,
		clientNameSurname,
		clientPhone,
		referance,
		givenDate,
		notificationToken,
	} = details;
	return (
		<List.Accordion
			title={licensePlate}
			key={licensePlate}
			left={(props) => <List.Icon {...props} icon="car" />}
		>
			{/* @ts-ignore */}
			<List.Item
				title={`Marka: ${brand}`}
				key={`${licensePlate}-${brand}`}
				left={(props) => <List.Icon {...props} icon="tag-multiple" />}
			/>
			{/* @ts-ignore */}
			<List.Item
				title={`Model: ${model}`}
				key={`${licensePlate}-${model}`}
				left={(props) => <List.Icon {...props} icon="tag" />}
			/>
			{/* @ts-ignore */}
			<List.Item
				title={`Müşteri: ${clientNameSurname}`}
				key={`${licensePlate}-${clientNameSurname}`}
				left={(props) => <List.Icon {...props} icon="account" />}
			/>
			{/* @ts-ignore */}
			<List.Item
				title={`Telefon: ${clientPhone}`}
				key={`${licensePlate}-${clientPhone}`}
				left={(props) => <List.Icon {...props} icon="phone" />}
				right={(props) => <List.Icon {...props} icon="call-made" />}
				onPress={() =>
					call({
						number: clientPhone.replace(' ', ''),
						prompt: false,
					}).catch(console.error)
				}
			/>
			{referance ? (
				// @ts-ignore
				<List.Item
					title={`Referans: ${referance}`}
					key={`${licensePlate}-${referance}`}
					left={(props) => <List.Icon {...props} icon="account-multiple" />}
				/>
			) : null}
			{/* @ts-ignore */}
			<List.Item
				title={`Verilen tarih: ${`${moment(givenDate).format('D MMMM YYYY')}`}`}
				key={`${licensePlate}-${givenDate}`}
				left={(props) => <List.Icon {...props} icon="calendar" />}
			/>
			{/* @ts-ignore */}
			<List.Item
				key={`${licensePlate}-controls`}
				right={() => (
					<>
						{/* @ts-ignore */}
						<Button
							uppercase={false}
							mode="outlined"
							icon="pencil"
							style={{ margin: 4 }}
							onPress={() =>
								navigation.navigate('Details', {
									day,
									edit: true,
									editThis: {
										licensePlate,
										brand,
										model,
										clientNameSurname,
										clientPhone,
										referance,
										givenDate,
										notificationToken,
									},
								})
							}
						>
							Düzenle
						</Button>
						{/* @ts-ignore */}
						<Button
							uppercase={false}
							style={{ margin: 4, width: 90 }}
							color={Colors.red500}
							mode="contained"
							icon="delete"
							onPress={() => {
								setContentToDelete(licensePlate);
								setDialogVisible(true);
							}}
						>
							Sil
						</Button>
					</>
				)}
			/>
		</List.Accordion>
	);
};

export default ListAccordion;
