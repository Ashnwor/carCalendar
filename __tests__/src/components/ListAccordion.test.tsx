import React from 'react';
import { render } from 'react-native-testing-library';

import ListAccordion from '../../../src/components/ListAccordion';

describe('<ListAccordion />', () => {
	it('renders without crashing', () => {
		const navigation: any = { navigate: jest.fn() };
		render(
			<ListAccordion
				navigation={navigation}
				day={new Date()}
				licensePlateName="test-name"
				details={{
					brand: 'test_brand',
					model: 'test_model',
					clientNameSurname: 'test_name_surname',
					clientPhone: '5555 555 55 55',
					referance: 'test_reference',
					givenDate: new Date(),
				}}
				functions={{
					setContentToDelete: jest.fn(),
					setDialogVisible: jest.fn(),
				}}
			/>,
		);
	});
});
