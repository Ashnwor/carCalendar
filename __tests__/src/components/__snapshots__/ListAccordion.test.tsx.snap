// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListAccordion /> renders without crashing 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        Array [
          Object {
            "padding": 8,
          },
          undefined,
        ],
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "margin": 8,
              "width": 40,
            },
            undefined,
          ]
        }
      >
        <Text />
      </View>
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Array [
                Object {
                  "fontSize": 16,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                },
                undefined,
              ],
            ]
          }
        >
          test-name
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "margin": 8,
            },
            undefined,
          ]
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
